### ISSUE TITLE
Feature request: FromJson format to local time

### ISSUE THREAD

**Issue by trtiger4520** [google/json_serializable.dart#1420]
Due to the need to display local time in the application

Want to add an option in the configuration file to allow users to choose whether to use local time

Achievable:
- Avoid manually changing to local time every time you use it
- Avoid modifying automatically generated files

Thanks

---

config:

``` yaml
targets:
  $default:
    builders:
      json_serializable:
        options:
          parse_local_time: true // default: false
```

from:

``` dart
part of 'example.dart';

Person _$PersonFromJson(Map<String, dynamic> json) => Person(
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      dateOfBirth: json['dateOfBirth'] == null
          ? null
          : DateTime.parse(json['dateOfBirth'] as String),
    );
```

to:

``` dart
part of 'example.dart';

Person _$PersonFromJson(Map<String, dynamic> json) => Person(
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      dateOfBirth: json['dateOfBirth'] == null
          ? null
          : DateTime.parse(json['dateOfBirth'] as String).toLocal(),
    );
```

**Comment by pattobrien** [comment]
> Avoid manually changing to local time every time you use it
> Avoid modifying automatically generated files

This is easily solvable by creating your own `JsonConverter` as shown:

```dart
import 'package:json_annotation/json_annotation.dart';

part 'date_converter.g.dart';

class LocalDateConverter implements JsonConverter<DateTime, String> {
  const LocalDateConverter();

  @override
  DateTime fromJson(String value) {
    return DateTime.parse(value).toLocal(); // implement your custom `fromJson` logic here
  }

  @override
  String toJson(DateTime value) {
    return value.toUtc().toIso8601String();  // implement your custom `toJson` logic here
  }
}

@JsonSerializable()
class Person {
  final String firstName;
  final String lastName;
  @LocalDateConverter() // annotate the field requiring custom serialization logic
  final DateTime dateOfBirth;

  const Person({
    required this.firstName,
    required this.lastName,
    required this.dateOfBirth,
  });

  factory Person.fromJson(Map<String, dynamic> json) => _$PersonFromJson(json);

  Map<String, dynamic> toJson() => _$PersonToJson(this);
}

```

Does that solve your problem?

**Comment by trtiger4520** [comment]
@pattobrien Thanks Reply

I understand that using `JsonConverter` can solve the conversion problem

However, my situation here is: the current project uses `swagger_dart_code_generator`, which is also part of the automatic generation. At that time, I looked at the contents of the files on both sides and found no better solution, so I came up with this idea.

I'll look for other solutions

---

in addition
I forked this repository and tried to adjust it, but found that `ToFromStringHelper` is placed in the final variable of the file. It seems that bringing config into the affected content will require adjusting a lot of structural issues.
